### PR TITLE
Updated dependencies.

### PR DESCRIPTION
**Pull request by jscarle**
- Updated all of the various NuGet packages to the latest versions.
- Replaced Moq with CleanMoq.

### CODE REVIEW

**Comment by jscarle** [comment]
@WojciechNagorski @Rob-Hague Ready for review and merge.

I replaced Moq with [CleanMoq](https://github.com/hassanhabib/CleanMoq). As I'm sure you're both aware of all of the drama surrounding Moq, CleanMoq was forked from Moq and will [continue to be maintained by Hassan Habib](https://twitter.com/jcarle/status/1691149981011292161).

File: src/Renci.SshNet/Renci.SshNet.csproj [hunk 1]
```diff
@@ -33,7 +34,7 @@
   </PropertyGroup>
 
   <ItemGroup Condition=" '$(TargetFramework)' == 'net462' or '$(TargetFramework)' == 'netstandard2.0' ">
-    <PackageReference Include="Microsoft.Bcl.AsyncInterfaces" Version="1.0.0" />
+    <PackageReference Include="Microsoft.Bcl.AsyncInterfaces" Version="8.0.0" />
```
Comment (WojciechNagorski) [inline src/Renci.SshNet/Renci.SshNet.csproj]: Can you revert this. You can check a history of this line.

Comment (jscarle) [reply src/Renci.SshNet/Renci.SshNet.csproj]: I reverted it back to version 1.0.0 and added a not so we don't try to update it again.

File: test/Renci.SshNet.Tests/Renci.SshNet.Tests.csproj [hunk 2]
```diff
@@ -8,11 +9,10 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.8.0" />
-    <PackageReference Include="MSTest.TestAdapter" Version="3.1.1" />
-    <PackageReference Include="MSTest.TestFramework" Version="3.1.1" />
-    <PackageReference Include="Moq" Version="4.18.4" />
```
Comment (WojciechNagorski) [inline test/Renci.SshNet.Tests/Renci.SshNet.Tests.csproj:14]: I think that `Moq` currently is clean, isn't it? On the other hand `CleanMoq` is just an unsupported and not maintained fork of  `Moq`. I would prefer to stick with "Moq" as long as it doesn't contain unsafe features.

Comment (jscarle) [reply test/Renci.SshNet.Tests/Renci.SshNet.Tests.csproj:14]: Well, depends on the version. The other has made it very clear he has no intention on abandoning his plan to implement SponsorLink in Moq.

Comment (jscarle) [reply test/Renci.SshNet.Tests/Renci.SshNet.Tests.csproj:14]: I reverted back to Moq.

**Review by WojciechNagorski — APPROVED** [review]
LGTM

**Pull request merged**